ibm/service/vpc/resource_ibm_is_share.go

```diff
@@ -0,0 +1,1077 @@
+	// tags, err := flex.GetGlobalTagsUsingCRN(meta, *share.CRN, "", isUserTagType)
+	// if err != nil {
+	// 	log.Printf(
+	// 		"Error getting shares (%s) tags: %s", d.Id(), err)
+	// }
+
+	accesstags, err := flex.GetGlobalTagsUsingCRN(meta, *share.CRN, "", isAccessTagType)
+	if err != nil {
+		log.Printf(
+			"Error getting shares (%s) access tags: %s", d.Id(), err)
+	}
+	d.Set(isFileShareAccessTags, accesstags)
+	// d.Set(isFileShareTags, tags)
+	if share.UserTags != nil {
+		if err = d.Set(isFileShareTags, share.UserTags); err != nil {
+			log.Printf(
+				"Error setting shares (%s) user tags: %s", d.Id(), err)
+		}
+	}
+
+	return nil
+}
+
+func resourceIbmIsShareUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcClient, err := meta.(conns.ClientSession).VpcV1BetaAPI()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	updateShareOptions := &vpcbetav1.UpdateShareOptions{}
+
+	updateShareOptions.SetID(d.Id())
+
+	hasChange := false
+
+	sharePatchModel := &vpcbetav1.SharePatch{}
+	if d.HasChange("name") {
+		name := d.Get("name").(string)
+		sharePatchModel.Name = &name
+		hasChange = true
+	}
+
+	if d.HasChange("size") {
+		size := int64(d.Get("size").(int))
+		sharePatchModel.Size = &size
+		hasChange = true
+	}
+
+	if d.HasChange("iops") {
+		iops := int64(d.Get("iops").(int))
+		sharePatchModel.Iops = &iops
+		hasChange = true
+	}
+
+	if d.HasChange("profile") {
+		old, new := d.GetChange("profile")
+		if old.(string) == "custom" {
+			if d.Get("iops").(int) != 0 {
+				log.Println("iops there")
+			} else {
+				log.Println("iops not there")
+			}
+		}
+		profile := new.(string)
+		sharePatchModel.Profile = &vpcbetav1.ShareProfileIdentity{
+			Name: &profile,
+		}
+		hasChange = true
+	}
+
+	if d.HasChange(isFileShareTags) {
+		var userTags *schema.Set
+		if v, ok := d.GetOk(isFileShareTags); ok {
+
+			userTags = v.(*schema.Set)
+			if userTags != nil && userTags.Len() != 0 {
+				userTagsArray := make([]string, userTags.Len())
+				for i, userTag := range userTags.List() {
+					userTagStr := userTag.(string)
+					userTagsArray[i] = userTagStr
+				}
+				schematicTags := os.Getenv("IC_ENV_TAGS")
+				var envTags []string
+				if schematicTags != "" {
+					envTags = strings.Split(schematicTags, ",")
+					userTagsArray = append(userTagsArray, envTags...)
+				}
+
+				sharePatchModel.UserTags = userTagsArray
+			}
+		}
+	}
+	/*
+		if d.HasChange("replication_cron_spec") {
+			replication_cron_spec := d.Get("replication_cron_spec").(string)
+			sharePatchModel.ReplicationCronSpec = &replication_cron_spec
+			hasChange = true
+		}
+	*/
+	if hasChange {
+
+		sharePatch, err := sharePatchModel.AsPatch()
+
+		if err != nil {
+			log.Printf("[DEBUG] SharePatch AsPatch failed %s", err)
+			return diag.FromErr(err)
+		}
+		updateShareOptions.SetSharePatch(sharePatch)
+		_, response, err := vpcClient.UpdateShareWithContext(context, updateShareOptions)
```

Comment:
Please add wait logic here for lifecycle_state to be stable

Reply:
<img width="651" alt="Screenshot 2023-02-09 at 3 17 05 PM" src="https://user-images.githubusercontent.com/78336632/217777276-5b5fad26-7ffd-4bdb-aabd-da5a38ff9cc8.png">